src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
 void EnKusa_Init(Actor* thisx, GlobalContext* globalCtx);
 void EnKusa_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnKusa_Update(Actor* thisx, GlobalContext* globalCtx);
+void func_80A9C164(Actor* thisx, GlobalContext* globalCtx);
 
-/*
-const ActorInit En_Kusa_InitVars = {
-    ACTOR_EN_KUSA,
-    ACTORTYPE_PROP,
-    FLAGS,
-    OBJECT_GAMEPLAY_KEEP,
-    sizeof(EnKusa),
-    (ActorFunc)EnKusa_Init,
-    (ActorFunc)EnKusa_Destroy,
-    (ActorFunc)EnKusa_Update,
-    NULL,
+void func_80A9AFA0(EnKusa* this, EnKusaActionFunc actionFunc);
+s32 func_80A9AFAC(EnKusa* this, GlobalContext* globalCtx, f32 arg2);
+void func_80A9B07C(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9B140(EnKusa* this);
+void func_80A9B174(Vec3f* this, f32 arg1);
+void func_80A9B1FC(EnKusa* this);
+void func_80A9B21C(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9B574(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9B630(Actor* thisx, GlobalContext* globalCtx);
+
+void func_80A9B7EC(EnKusa* this);
+void func_80A9B810(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9B89C(EnKusa* this);
+void func_80A9B8D8(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9BA98(EnKusa* this);
+void func_80A9BAD8(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9BBB0(EnKusa* this);
+void func_80A9BC1C(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9BEAC(EnKusa* this);
+void func_80A9BEFC(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9BF30(EnKusa* this, EnKusaActionFunc actionFunc);
+void func_80A9BF3C(EnKusa* this);
+void func_80A9BFA8(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9C00C(EnKusa* this);
+void func_80A9C068(EnKusa* this, GlobalContext* globalCtx);
+void func_80A9C164(Actor* thisx, GlobalContext* globalCtx);
+
+s16 D_80A9C1D0[] = { 0x00000000 };
+
+s16 D_80A9C1D4[] = { 0x00000000 };
+
+s16 D_80A9C1D8[] = { 0x00000000 };
+
+s16 D_80A9C1DC[] = { 0x00000000 };
```

Comment:
Tried to change things, but I couldn't get it to match, I might need to know more along the lines of what you were thinking needs to be changed below.

Reply:
after you make this change, you need to change the accesses of these symbols to not be a dereference. just treat is as a normal variable

Reply:
nvm im assuming you did that, the real issue is that MathAproxS is expecting a pointer to the value, not the value itself.
Before you had no & symbol cause its implied for arrays, but with it not an array anymore you had to add it.

I figured this out by using first_diff and seeing what function the issue was in, and saw an lh instead of an addiu